Packages/ClientRuntime/Tests/SerializationTests/SerializationUtilsTests/TimestampSerdeUtilsTests.swift

```diff
@@ -0,0 +1,161 @@
+        let encoder = JSONEncoder()
+        
+        struct Container: Encodable {
+            let timestamp: Date
+            enum CodingKeys: String, CodingKey {
+                case timestamp
+            }
+            func encode(to encoder: Encoder) throws {
+                var container = encoder.container(keyedBy: CodingKeys.self)
+                try container.encodeTimestamp(
+                    timestamp,
+                    format: .dateTime,
+                    forKey: .timestamp
+                )
+            }
+        }
+        let container = Container(timestamp: testDateWithFractionalSeconds)
+        let data = try encoder.encode(container)
+        let dataAsString = String.init(data: data, encoding: .utf8)!
+        XCTAssertEqual(dataAsString, "{\"timestamp\":\"1991-05-04T10:12:10.123Z\"}")
+    }
+    
+    func test_encodeTimeStamp_forSingleValueContainer_returnsExpectedValue() throws {
+        let encoder = JSONEncoder()
+        
+        struct Container: Encodable {
+            let timestamp: Date
+            func encode(to encoder: Encoder) throws {
+                var container = encoder.singleValueContainer()
+                try container.encodeTimestamp(timestamp, format: .dateTime)
+            }
+        }
+        let container = Container(timestamp: testDateWithFractionalSeconds)
+        let data = try encoder.encode(container)
+        let dataAsString = String.init(data: data, encoding: .utf8)!
+        XCTAssertEqual(dataAsString, "\"1991-05-04T10:12:10.123Z\"")
+    }
+    
+    // MARK: - Decoding Tests
+    
+    func test_decodeTimestamp_returnsExpectedValue() throws {
+        struct Container: Decodable {
+            let timestamp: Date
+            static var format: TimestampFormat = .dateTime
+            enum CodingKeys: String, CodingKey {
+                case timestamp
+            }
+            init(from decoder: Decoder) throws {
+                let container = try decoder.container(keyedBy: CodingKeys.self)
+                self.timestamp = try container.decodeTimestamp(Self.format, forKey: .timestamp)
+            }
+        }
+        
+        let subjects: [(TimestampFormat, String, Date)] = [
+            (.epochSeconds, "{\"timestamp\":\(testDateWithFractionalSeconds.timeIntervalSince1970)}", testDateWithFractionalSeconds),
+            (.epochSeconds, "{\"timestamp\":\(testDateWithoutFractionalSeconds.timeIntervalSince1970)}", testDateWithoutFractionalSeconds),
+            (.dateTime, "{\"timestamp\":\"1991-05-04T10:12:10.123Z\"}", testDateWithFractionalSeconds),
+            (.dateTime, "{\"timestamp\":\"1991-05-04T10:12:10Z\"}", testDateWithoutFractionalSeconds),
+            (.httpDate, "{\"timestamp\":\"Sat, 04 May 1991 10:12:10.123 GMT\"}", testDateWithFractionalSeconds),
+            (.httpDate, "{\"timestamp\":\"Sat, 04 May 1991 10:12:10 GMT\"}", testDateWithoutFractionalSeconds)
```

Comment:
should there be a test for XML Encode and Decoder?

Reply:
This test should be decoder agnostic. The purpose of this test is to test the `decodeTimestamp(,forKey:)` function. Unfortunately, we need to use some decoder to do that, in this case its the JSON decoder. But really, the specific decoder shouldn't matter here.